File: app/src/main/java/com/ngengs/android/popularmovies/apps/utils/NetworkHelpers.java
package com.ngengs.android.popularmovies.apps.utils;

import com.ngengs.android.popularmovies.apps.globals.Values;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by ngengs on 7/4/2017.
 */

@SuppressWarnings({"WeakerAccess"})
public class NetworkHelpers {
    public static Retrofit provideRetrofit() {
        return new Retrofit.Builder()
                .baseUrl(Values.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.createAsync())
                .build();
    }

    public static MoviesAPI provideAPI() {
        return provideRetrofit().create(MoviesAPI.class);
    }
}

File: app/src/main/java/com/ngengs/android/popularmovies/apps/utils/NetworkHelpers.kt
package com.ngengs.android.popularmovies.apps.utils

import com.ngengs.android.popularmovies.apps.globals.Values
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by rizky.kharisma on 09/11/22.
 * @ngengs
 */
object NetworkHelpers {
    private fun provideRetrofit(): Retrofit {
        return Retrofit.Builder()
            .baseUrl(Values.URL_BASE)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.createAsync())
            .build()
    }

    fun provideAPI(): MoviesAPI {
        return provideRetrofit().create(MoviesAPI::class.java)
    }
}